communicator/mainwindow.cpp

```diff
@@ -0,0 +1,14 @@
+#include "mainwindow.h"
+#include "./ui_mainwindow.h"
+
+MainWindow::MainWindow(QWidget *parent)
+    : QMainWindow(parent)
+    , ui(new Ui::MainWindow)
+{
+    ui->setupUi(this);
+}
+
+MainWindow::~MainWindow()
+{
+    delete ui;
+}
```

Comment:
maybe use smart pointers?

Reply:
we cannot, Qt is very restrictive in case of their own API, https://stackoverflow.com/questions/59634605/using-stdunique-ptr-on-qt

Reply:
Ok, I have no experience with qt and didn't know